src/libraries/System.Threading.AccessControl/src/System/Threading/EventWaitHandleAcl.cs

```diff
@@ -56,32 +58,120 @@ public static unsafe EventWaitHandle Create(bool initialState, EventResetMode mo
+                case OpenExistingResult.NameInvalid:
+                    throw new WaitHandleCannotBeOpenedException(SR.Format(SR.Threading_WaitHandleCannotBeOpenedException_InvalidHandle, name));
 
-            if (handle.IsInvalid)
+                case OpenExistingResult.PathNotFound:
+                    throw new IOException(SR.Format(SR.IO_PathNotFound_Path, name));
+
+                case OpenExistingResult.Success:
+                default:
+                    Debug.Assert(result != null, "result should be non-null on success");
+                    return result;
+            }
+        }
+
+        /// <summary>
+        /// Tries to open a specified named event wait handle, if it already exists, applying the desired access rights, and returns a value that indicates whether the operation succeeded.
+        /// </summary>
+        /// <param name="name">The name of the event wait handle to be opened. If it's prefixed by "Global", it refers to a machine-wide event wait handle. If it's prefixed by "Local", or doesn't have a prefix, it refers to a session-wide event wait handle. Both prefix and name are case-sensitive.</param>
+        /// <param name="rights">The desired access rights to apply to the returned event wait handle.</param>
+        /// <param name="result">When this method returns <see langword="true" />, contains an object that represents the named event wait handle if the call succeeded, or <see langword="null" /> otherwise. This parameter is treated as uninitialized.</param>
+        /// <returns><see langword="true" /> if the named event wait handle was opened successfully; otherwise, <see langword="false" />.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="name"/> is <see langword="null" /></exception>
+        /// <exception cref="ArgumentException"><paramref name="name"/> is an empty string.</exception>
+        /// <exception cref="IOException">A Win32 error occurred.</exception>
+        /// <exception cref="UnauthorizedAccessException">The named event wait handle exists, but the user does not have the security access required to use it.</exception>
+        public static bool TryOpenExisting(string name, EventWaitHandleRights rights, [NotNullWhen(returnValue: true)] out EventWaitHandle? result) =>
+            OpenExistingWorker(name, rights, out result) == OpenExistingResult.Success;
+
+        private static OpenExistingResult OpenExistingWorker(string name, EventWaitHandleRights rights, out EventWaitHandle? result)
+        {
+            if (name == null)
             {
-                handle.SetHandleAsInvalid();
+                throw new ArgumentNullException(nameof(name));
+            }
+            if (name.Length == 0)
+            {
+                throw new ArgumentException(SR.Argument_EmptyName, nameof(name));
+            }
+            if (rights < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(rights), SR.ArgumentOutOfRange_Enum);
```

Comment:
Makes sense. We also don't check for the values of `rights` in the .NET Core overloads. I'll remove the checks so Windows can do it for us.

Reply:
Just so you know: I originally wrote a unit test to throw `ArgumentOutOfRangeException` if the `rights` value was below 0.
If I remove this check, the same unit test will throw based on the returned error value from Windows. The error is now:

`System.Threading.WaitHandleCannotBeOpenedException : No handle of the given name exists.`

Reply:
I think that error return code is only because there is actually no synchronization object with that name. `-1` translates to `GENERIC_READ` rights, which maps to specific rights for the type of object, and seems to work for opening an existing synchronization object. I'm not sure if there is such a thing as invalid rights, if so I would expect a return code of `ERROR_INVALID_PARAMETER`, likely bits that don't affect the rights of the object would just be ignored. I would probably just exclude the tests for invalid rights.

Reply:
>_you should run your unit tests against both .NET Framework and .NET 6_

Will do. I'm building for .NET Framework right now.

> _I would probably just exclude the tests for invalid rights._

Or as a last attempt, let me use `Int32.MinValue`. I'll let you know what happens.

Reply:
Update: `Int32.MinValue` got the same result as if I was using `-1` (`WaitHandleCannotBeOpenedException: No handle of the given name exists`). So as you explained, @kouvel, this unit test might not be useful since the exception is unrelated to the invalid enum value. I'll remove the unit tests.

Reply:
Update 2: I ran the tests locally against .NET Framework and they passed.